src/middlewared/middlewared/plugins/idmap.py

```diff
@@ -693,10 +714,33 @@ async def sid_to_name(self, sid_str):
         return rv
 
     @private
-    async def name_to_sid(self, name):
-        wb = await run([SMBCmd.WBINFO.value, '--name-to-sid', name], check=False)
+    async def unixid_to_sid(self, data):
+        """
+        Samba generates SIDs for local accounts that lack explicit mapping in
+        passdb.tdb or group_mapping.tdb with a prefix of S-1-22-1 (users) and
+        S-1-22-2 (groups). This is not returned by wbinfo, but for consistency
+        with what appears when viewed over SMB protocol we'll do the same here.
+        """
+        unixid = data.get("id")
+        id = IDType[data.get("id_type", "GROUP")]
+        wb = None
```

Comment:
Is there a possibility that the `wb` variable returns errno != 0 but has no data in it? If that happens, the `wb.decode().strip()` will fail because it will be of None type, right?

Maybe before returning wb.decode() check to make sure it's `isinstance(wb, bytes)`?

Reply:
If thats the case it will be empty string `b''`, never None as far as i can tell

we could just remove wb = None since its superfluous